dags/multipod/configs/mxla_collective_config.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Utilities to construct configs for maxtext DAG."""
+
+from xlml.apis import gcp_config, metric_config, task, test_config
+from dags import test_owner, gcs_bucket
+from dags.multipod.configs import common
+from dags.vm_resource import TpuVersion, Project, RuntimeVersion
+import datetime
+
+PROJECT_NAME = Project.CLOUD_ML_AUTO_SOLUTIONS.value
+RUNTIME_IMAGE = RuntimeVersion.TPU_UBUNTU2204_BASE.value
+
+
+def get_mxla_collective_config(
+    tpu_version: TpuVersion,
+    tpu_cores: int,
+    tpu_zone: str,
+    time_out_in_min: int,
+    test_name: str,
+    bytes_to_transfer: int,
+    project_name: str = PROJECT_NAME,
+    runtime_version: str = RUNTIME_IMAGE,
+    network: str = "default",
+    subnetwork: str = "default",
+    is_tpu_reserved: bool = True,
+    automated_test: bool = True,
+    num_slices: int = 1,
+) -> task.TpuQueuedResourceTask:
+  job_gcp_config = gcp_config.GCPConfig(
+      project_name=project_name,
+      zone=tpu_zone,
+      dataset_name=metric_config.DatasetOption.XLML_DATASET,
+      dataset_project=project_name,
+      composer_project=project_name,
+  )
+
+  current_time = datetime.datetime.now()
+  current_date = current_time.strftime("%Y-%m-%d")
+  current_datetime = current_time.strftime("%Y-%m-%d-%H-%M-%S")
+
+  trigger = "automated" if automated_test else "manual"
+  base_output_directory = f"{gcs_bucket.XLML_OUTPUT_DIR}/multipod/mxla/nightly/{trigger}/{current_date}/{num_slices}slice-V{tpu_version.value}_{tpu_cores}-mxla-collective-{bytes_to_transfer}transferBytes-{current_datetime}"
+
+  test_platform = common.Platform.GCE
+  set_up_cmds = common.setup_mxla_collective()
+  run_model_cmds = (
+      (
+          "cd /tmp/mxla_collective &&"
+          f" sudo bash run_mxla_collective_benchmark_gcp.sh BYTES_TO_TRANSFER={bytes_to_transfer} NUM_STEPS=100 SLICES={num_slices} ACCELERATOR_TYPE=v{tpu_version.value}_{tpu_cores} GCS_PATH={base_output_directory}"
+          " exit $(cat /tmp/benchmark_status.txt)"
```

Comment:
Where does `benchmark_status.txt` come from?

Reply:
It will be generated from `run_mxla_collective_benchmark_gcp.sh` and store in TPU VM.

Reply:
Does `benchmark_status.txt` contain the results of the benchmark and does it get uploaded to GCS?

Reply:
Here is an example of the results:
https://screenshot.googleplex.com/4R8dhk8yfNoogTV

`benchmark_status.txt` will only contain the exit status of this workload to tell us if this workload passed or failed.